app/models/experimentation/notifications_experiment.rb

```diff
@@ -141,10 +137,50 @@ def reporting_data(patient, date)
       }
     end
 
+    def notification_result(notification_id)
+      communications = Communication.where(notification_id: notification_id)
+      successful_delivery = communications.with_delivery_detail.find_by(
+        delivery_detail: {result: [:read, :delivered, :sent]}
+      )
```

Comment:
I haven't moved this to `TwilioDeliveryDetail`/communication since the criteria for success is slightly different from the methods in `TwilioDeliveryDetail`. We consider `sent` a success (for SMS), whereas it is considered `in_progress` [here](https://github.com/simpledotorg/simple-server/blob/6d44b863600a078dfffe5063554b658e60893fe0/app/models/twilio_sms_delivery_detail.rb#L24). I have tried to avoid changing those methods since they are being used elsewhere.